### PR TITLE
use latest 3dt

### PR DESCRIPTION
**Pull request by darkonie**
add snapshot functionality
https://github.com/dcos/3dt/pull/28

### CODE REVIEW

**Comment by darkonie** [comment]
@cmaloney this is ready for review

File: test_util/integration_test.py [hunk 1]
```diff
@@ -1422,3 +1424,159 @@ def test_mesos_agent_role_assignment(cluster):
...
+
+        # check `is_running` attribute for each host. All of them must be False
+        for _, attributes in response.items():
+            assert not attributes['is_running']
+
+    wait_for_job()
+
+    # the job should be complete at this point.
+    # check the listing for a zip file
+    snapshots = _get_snapshot_list(cluster)
```
Comment (cmaloney) [inline test_util/integration_test.py]: Is there some way we can test that the snapshot has some of the data we want in it?

File: test_util/integration_test.py [hunk 2]
```diff
@@ -1422,3 +1424,159 @@ def test_mesos_agent_role_assignment(cluster):
...
+    expected_master_files = ['dcos-mesos-master.service.gz'] + expected_common_files
+
+    # for agent host
+    expected_agent_files = ['dcos-mesos-slave.service.gz'] + expected_common_files
+
+    # for public agent host
+    expected_public_agent_files = ['dcos-mesos-slave-public.service.gz'] + expected_common_files
+
+    try:
+        tmp_dir = tempfile.mkdtemp()
```
Comment (cmaloney) [inline test_util/integration_test.py]: Python has https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory to make it so you don't have to hand write the cleanup it in the try / except here.

File: test_util/integration_test.py [hunk 3]
```diff
@@ -1422,3 +1424,159 @@ def test_mesos_agent_role_assignment(cluster):
...
+        download_base_url = '/system/health/v1/report/snapshot/serve'
+        for snapshot in snapshots:
+            snapshot_full_location = os.path.join(tmp_dir, snapshot)
+            with open(snapshot_full_location, 'wb') as f:
+                r = cluster.get(path=os.path.join(download_base_url, snapshot), stream=True)
+                for chunk in r.iter_content(1024):
+                    f.write(chunk)
+
+            # validate snapshot zip file.
+            print(snapshot_full_location)
```
Comment (cmaloney) [inline test_util/integration_test.py]: print() is used here but logging() is used above. Should be consistent. logging is the most frequently used.

File: test_util/integration_test.py (hunk 1 above)
Comment (darkonie) [inline test_util/integration_test.py]: fixed.
response['extra']['snapshot_name'] contains the snapshot unique filename.

File: test_util/integration_test.py (hunk 2 above)
Comment (darkonie) [inline test_util/integration_test.py]: done

File: test_util/integration_test.py (hunk 3 above)
Comment (darkonie) [inline test_util/integration_test.py]: done

**Comment by darkonie** [comment]
@cmaloney this is ready for round 2 code review. addressed you comments.

**Comment by cmaloney** [comment]
Waiting for an out of band go code review from one of our go experts

**Pull request merged**